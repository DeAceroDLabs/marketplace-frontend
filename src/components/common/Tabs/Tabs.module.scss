.tabs {
  display: flex;
  justify-content: space-between;
  overflow: auto;
  -ms-overflow-style: none;
  scrollbar-width: none;
}

.tabs::-webkit-scrollbar {
  display: none;
}
