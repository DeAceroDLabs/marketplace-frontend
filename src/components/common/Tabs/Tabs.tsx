import { useState } from "react";
import Tab from "./Tab/Tab";
import styles from "./Tabs.module.scss";

interface TabsProps {
  variant?: "primary" | "secondary";
  tabsTitles: string[];
}

const Tabs: React.FC<TabsProps> = ({ variant = "primary", tabsTitles }) => {
  const [activeTab, setactiveTab] = useState(tabsTitles[0]);

  const setActive = (tabTitle: string) => {
    setactiveTab(tabTitle);
  };

  const tabs = tabsTitles.map((title: string) => {
    return (
      <Tab
        key={title}
        variant={variant}
        title={title}
        active={activeTab === title ? "active" : ""}
        onSelect={setActive}
      />
    );
  });
  return <div className={styles.tabs}>{tabs}</div>;
};

export default Tabs;
