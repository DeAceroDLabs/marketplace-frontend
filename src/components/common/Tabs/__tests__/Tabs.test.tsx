import { TabItem } from "components/common/common.types";
import renderer from "react-test-renderer";
import Tabs from "../Tabs";

const mockTabs: TabItem[] = [
  {
    id: 1,
    title: "tab 1",
  },
  {
    id: 2,
    title: "tab 2",
  },
];

const TabsMockProps = {
  options: mockTabs,
  onSelectTab: () => jest.fn(),
};
it("renders Tabs with no issue", () => {
  const component = renderer.create(<Tabs {...TabsMockProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
