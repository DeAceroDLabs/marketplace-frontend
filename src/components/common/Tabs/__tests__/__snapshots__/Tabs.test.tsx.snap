// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tabs with no issue 1`] = `
<div
  className="tabs"
>
  <button
    className="tab primary active"
    onClick={[Function]}
  >
    tab 1
  </button>
  <button
    className="tab primary "
    onClick={[Function]}
  >
    tab 2
  </button>
</div>
`;
